polyfill/lib/ecmascript.mjs

```diff
@@ -255,6 +255,7 @@ export const ES = ObjectAssign({}, ES2020, {
     const year = ES.ToInteger(yearString);
     const month = ES.ToInteger(match[2] || match[4]);
     const day = ES.ToInteger(match[3] || match[5]);
+    if (timeRequired && match[6] === undefined) throw new RangeError(`invalid ISO 8601 string: ${isoString}`);
```

Comment:
Hey @ptomato - Whoops, I missed reviewing this originally. Sorry for coming late to this party!  After doing a lot of surgery recently on parsing methods, I think a better fix here would be to move the exception into `ParseTemporalTimeString` instead, which enables a clearer type-specific error message, e.g. `time is missing from ISO string: ${isoString}`. 

This would be also be more consistent with the pattern we now use in `ParseTemporalInstantString`, `ParseTemporalZonedDateTimeString`, `ParseTemporalYearMonthString`, `ParseTemporalMonthDayString`, and `ParseTemporalTimeZoneString`, where `ParseISODateTime` handles parsing only while type-specific validation happens in the higher-level functions.  What do you think?

Reply:
Or, more likely, probably I missed a subtle reason why that same pattern won't work here.

Reply:
Fixed in #1986.